cmd/snap-confine/seccomp-support.c

```diff
@@ -39,17 +45,27 @@ int sc_apply_seccomp_bpf(const char *filter_profile)
 			 filter_profile_dir, filter_profile);
 
 	// load bpf
-	char bpf[32 * 1024];
+	char bpf[MAX_BPF_SIZE];
 	int fd = open(profile_path, O_RDONLY);
 	if (fd < 0)
 		die("cannot read %s", profile_path);
+	struct stat stat_buf;
+
+	if (fstat(fd, &stat_buf) < 0)
+		die("cannot stat %s", profile_path);
+	if (stat_buf.st_size > MAX_BPF_SIZE)
+		die("profile %s is too big %lu", profile_path,
+		    stat_buf.st_size);
```

Comment:
Note, stat() is a little weird in that while you pass it an fd, the size is checked at the time of the call and does not return the size from when the file was open()ed. Therefore, this is technically racy with the open but that's ok with how you are using it (ie, as a quick check for convenient error reporting). You are correctly giving 'read()' the maximum size which is MAX_BPF_SIZE so there is no need to check num_read is too big (though note that between the stat and the read the file could've been made bigger and therefore the read truncated).

In thinking about this and the issues with read() I suggest:
```
FILE *fp;
unsigned char bpf[MAX_BPF_SIZE + 1]; // acount for EOF

fp = fopen(profile_path, "rb");
if (fp == NULL) {
	die("cannot read %s", profile_path);

// set 'size' to 1 to get bytes transferred
size_t num_read = fread(bpf, 1, sizeof(bpf), fp);

if (ferror(fp) != 0)
	die("cannot fread() %s", profile_name);
else if (feof(fp) == 0)
	die("profile %s exceeds %zu bytes", profile_path, sizeof(bpf));

fclose(fp);
debug("read %zu bytes from %s", num_read, profile_name);
```